tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -720,31 +764,52 @@ def from_generator(generator, output_types, output_shapes=None, args=None):
         `iter()` protocol. If `args` is not specified, `generator` must take no
         arguments; otherwise it must take as many arguments as there are values
         in `args`.
-      output_types: A nested structure of `tf.DType` objects corresponding to
-        each component of an element yielded by `generator`.
+      output_types: (Optional.) A nested structure of `tf.DType` objects
```

Comment:
My suggestion is to deprecate `output_types` and `output_shapes` in favor of the more general `output_spec`. The documentation should state this.

Reply:
How exactly do I deprecate it, I mean - just change the docs, or mark in some wrapper, etc?. They I did it was to make it backward compatible. I would of course love it not to support to different ways of calling too.

Reply:
Here is an example: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/ops/dataset_ops.py#L205-L206

Reply:
The argument itself does not go away to preserve backwards compatibility. We will remove it when TF 3.0 is released.

Reply:
done